src/commands/logStream/logStreamRequest.ts

```diff
@@ -0,0 +1,129 @@
+                const abortController: AbortController = new AbortController();
+
+                const genericClient: ServiceClient = await createGenericClient(context, undefined);
+                const headers = createHttpHeaders({
+                    authorization: `Bearer ${token.token}`
+                })
+
+                const logsResponse = await genericClient.sendRequest(createPipelineRequest({
+                    method: "GET",
+                    url: endpoint,
+                    abortSignal: abortController.signal,
+                    headers,
+                    streamResponseStatusCodes: new Set<number>([200])
+                }));
+
+                await new Promise<void>((onLogStreamEnded: () => void, reject: (err: Error) => void): void => {
+                    const newLogStream: ILogStream = {
+                        dispose: (): void => {
+                            logsResponse.readableStreamBody?.removeAllListeners();
+                            abortController.abort();
+                            outputChannel.show();
+                            outputChannel.appendLine(localize('logStreamDisconnected', 'Disconnected from log-streaming service.'));
+                            newLogStream.isConnected = false;
+                            void onLogStreamEnded();
+                        },
+                        isConnected: true,
+                        outputChannel: outputChannel,
+                        data: {
+                            revision: context.revision?.name,
+                            replica: context.replica?.name,
+                            container: context.container?.name,
+                            containerApp: context.containerApp.name,
+                        }
+                    };
+
+                    logsResponse.readableStreamBody?.on('data', (chunk: Buffer | string) => {
+                        outputChannel.append(chunk.toString());
+                    }).on('error', (err: Error) => {
+                        newLogStream.isConnected = false;
+                        outputChannel.show();
+                        outputChannel.appendLine(localize('logStreamError', 'Error connecting to log-streaming service:'));
+                        outputChannel.appendLine(parseError(err).message);
+                        reject(err);
+                    }).on('complete', () => {
+                        newLogStream.dispose();
+                    });
+                    logStreams.set(logStreamId, newLogStream);
+                    onLogStreamCreated(newLogStream);
+                });
+            });
+        });
+    }
+}
+
+export async function disconnectLogStreaming(context: IStreamLogsContext): Promise<void> {
+    const allStreams = context.logStreamToStop ? [context.logStreamToStop] : getActiveLogStreams(context);
+
+    for (const streams of allStreams.values()) {
```

Comment:
What happens if there are no active log streams? We should probably throw an error. There's another spot `getActiveLogStreams`  is used (and an error is thrown) so maybe it makes sense to just have the error be thrown from that function?

Reply:
`StreamListStep` currently throws an error if there are no active streams. This is called before `disconnectLogStreaming ` so it should already be thrown if there are no active streams.

Reply:
That makes sense. 

I'd personally still prefer to have the error be thrown in `getActiveLogStreams` though.  I know it's probably unlikely, but in the event that we ever have to call `disconnectLogStreaming` from somewhere else, it'd be good to have the error behavior consistent.